### ISSUE TITLE
Training question

### ISSUE THREAD

**Issue by 893018991** [DerThomy/AAA-Gaussians#3]
Hello,  

When attempting to reproduce the experimental results, I was unable to obtain outcomes consistent with those reported in the paper.

Specifically, I trained using the Train dataset from Tanks & Temples, while the configuration file utilized the default aaa.json in the code

{  
    "culling_settings": {  
        "hierarchical_4x4_culling": true,  
        "rect_bounding": true,  
        "tight_opacity_bounding": true,  
        "tile_based_culling": true  
    },  
    "load_balancing": true,  
    "proper_ewa_scaling": true,  
    "eval_3D": true,  
    "new_aabb": true,  
    "sort_settings": {  
        "queue_sizes": {  
            "per_pixel": 4,  
            "tile_2x2": 8,  
            "tile_4x4": 64  
        },  
        "sort_mode": 3,  
        "sort_order": 3  
    }  
}

The results I obtained by running metrics.py are as follows:

{  
 "ours_30000": {  
  "SSIM": 0.7659546136856079,  
  "PSNR": 20.55414390563965,  
  "LPIPS": 0.2833644151687622,  
  "FLIPS": 0.28712546825408936,  
  "NUM": 182686  
 }  
}

In the paper, the results under the same conditions were:
  SSIM: 0.833 
  PSNR": 21.73  
  LPIPS": 0.184

Is this inconsistency related to the parameters in the configuration file aaa.json?  When using different datasets, do the parameters in the configuration file aaa.json need to be adjusted?

**Comment by DerThomy** [comment]
Hi, thanks for reaching out!

No, the parameters used for the paper evaluation do not change for different datasets. To verify we just reran our full_eval.py script with aaa.json on the Train scene and got the following results:

PSNR: 21.85 vs 21.73
SSIM: 0.834 vs. 0.833
LPIPS: 0.185 vs. 0.184

Could you maybe elaborate on your training and testing setup?

**Comment by 893018991** [comment]
Hello, thank you for your reply!

for our training and testing setup, Some things that need to be explained are:
   1.   I used a single dataset for training and did not use the aaa_gs.zip dataset you provided because it is too large. (I have just started downloading it now)

   2.   At the beginning of the training and testing, I did not set the cap_max parameter according to the parameters in full_ eval (according to your reminder, I have already changed it).

   3.   Here are the instructions I used for the training and testing I just conducted:

# python train.py --splatting_config configs/aaa.json -s ./Data/tandt_db/tandt/train --model_path ./trainout/capmax1026508/t1 --cap_max 1026508 --eval
# python render.py -m ./trainout/capmax1026508/t1
# python metrics.py -m ./trainout/capmax1026508/t1

For the result (iteration 30000) :   
SSIM :    0.7917013
PSNR :   20.6712608
LPIPS:    0.2316017
FLIP :    0.2912954
NUM  :   1026508


The structure of the dataset is as follows (The dataset comes from the tandt_db dataset provided by Inria 3D Gaussian Splatting.）:
<img width="299" height="245" alt="Image" src="https://github.com/user-attachments/assets/036cf0a6-55e7-435d-b483-eba578a28882" />

Is the inconsistency of the results related to the inconsistency of the dataset?

**Comment by DerThomy** [comment]
Hi, thanks for clarifying!

No the dataset should be fine, we used the same one.
Your call to the training script seems fine, the only thing I notice is that you do not pass --eval and --skip_train to render.py. --skip_train should not make a difference but since you dont need the training view renderings for evaluation it saves some time. Without --eval however, no test cams are loaded and rendered.

**Comment by 893018991** [comment]
Hello, thank you for your reply!

I have encountered a new problem, I encountered a new problem where I trained using the/nerf_360/cycle dataset.

The training instructions are as follows:

python train.py --splatting_config configs/aaa.json -s ../data/nerf_360/bicycle --model_path ./trainout/capmax6131954/t1 --cap_max 6131954 --eval

At the end of the training, the following results were obtained：

Training progress: 100%|████████████████████| 30000/30000 [1:00:30<00:00,  8.26it/s, Loss=0.0812714]

[ITER 30000] Evaluating test: L1 0.037966637387871746 PSNR 25.462158432006838 [11/12 12:35:32]

[ITER 30000] Evaluating train: L1 0.03596721887588501 PSNR 24.24933547973633 [11/12 12:35:32]

[ITER 30000] Saving Gaussians [11/12 12:35:32]
Segmentation fault (core dumped)

<img width="302" height="367" alt="Image" src="https://github.com/user-attachments/assets/de119fd7-fc26-4cbc-9c7e-1a9f22ff91af" />

The result of the 7000th iteration can be preserved, but the result of the 30000th iteration cannot be saved. Do you know the reason for this?  Have you ever encountered such a situation?

**Comment by DerThomy** [comment]
Hey, sorry we did not encounter this issue. You can try running in debug mode to pin point the issue.

**Comment by 893018991** [comment]
Alright, thank you again for your answer!
